ruddock/modules/users/helpers.py

```diff
@@ -33,11 +33,40 @@ def get_memberlist(search_type):
     """.format(tables))
   return flask.g.db.execute(query).fetchall()
 
-def get_user_info(username):
-  """Retrieves a user's info."""
+def get_office_info(user_id):
+  """Procedure to get a user's officer info."""
+  query = sqlalchemy.text("""
+    SELECT office_name, start_date, end_date
+    FROM office_assignments NATURAL JOIN members NATURAL JOIN offices
+    WHERE user_id = :u
+    ORDER BY start_date, end_date, office_name
+  """)
+  return flask.g.db.execute(query, u=user_id).fetchall()
+
+def get_user_id_from_username(username):
+  """
+  Given a user's username, returns their user_id. Returns None if no user
+  with this username exists.
+  """
+  query = sqlalchemy.text('SELECT user_id FROM users WHERE username=:u')
+  result = flask.g.db.execute(query, u=username).first()
+  if result is not None:
+    return result['user_id']
+  else:
+    return None
+
+def get_member_info(user_id):
```

Comment:
I tried editing each field, they all work except membership type. Add member_type to this query and everything works :)